ftw/upgrade/workflow.py

```diff
@@ -166,8 +166,14 @@ def lookup_objects(self, types):
         query = {'portal_type': types}
         brains = tuple(catalog.unrestrictedSearchResults(query))
 
-        lookup = lambda brain: portal.unrestrictedTraverse(brain.getPath())
-        generator = SizedGenerator((lookup(brain) for brain in brains),
+        def lookup(brain):
+            try:
+                return portal.unrestrictedTraverse(brain.getPath())
+            except KeyError:
+                return None
+
+        generator = (lookup(brain) for brain in brains)
```

Comment:
I see two problems here:
1. We are not uncataloging missing objects.
2. This line does not filter `None` elements, so the length of the SizedGenerator in the next line is too big when there are broken brains.

Could we move `catalog_unrestricted_get_object` from the step class into a place where we can reuse it here, so that we can use the same code?
I'd then use the same code as in #146:
```python
generator = (self.catalog_unrestricted_get_object(brain) for brain in brains)
generator = (obj for obj in generator if obj is not None)
return SizedGenerator(generator, len(brains))
```